app/models/message_draft.rb

```diff
@@ -153,45 +154,54 @@ def create_form_object
     )
   end
 
-  private
+  # TODO remove UPVS stuff from core domain
+  def validate_form
+    raise "Disallowed form" unless ::Upvs::ServiceWithFormAllowRule.form_services(all_metadata).any?
 
-  def validate_with_message_template
-    template&.validate_message(self)
-  end
+    raise "Missing XSD schema" unless upvs_form&.xsd_schema
 
-  def validate_metadata
-    all_message_metadata = if template&.metadata.present?
-     metadata.merge(template.metadata)
-    else
-     metadata
-    end
+    form_content = form&.unsigned_content
 
-    errors.add(:metadata, "No recipient URI") unless all_message_metadata["recipient_uri"].present?
-    errors.add(:metadata, "No posp ID") unless all_message_metadata["posp_id"].present?
-    errors.add(:metadata, "No posp version") unless all_message_metadata["posp_version"].present?
-    errors.add(:metadata, "No message type") unless all_message_metadata["message_type"].present?
-    errors.add(:metadata, "No correlation ID") unless all_message_metadata["correlation_id"].present?
-    errors.add(:metadata, "Correlation ID must be UUID") unless all_message_metadata["correlation_id"]&.match?(Utils::UUID_PATTERN)
-  end
+    if form_content
+      document = Nokogiri::XML(form_content)
+      form_errors = document.errors
 
-  def validate_metadata_with_template
-    errors.add(:metadata, :no_template) unless metadata&.dig("template_id").present?
-  end
+      document = Nokogiri::XML(document.children.first.children.first.children.first.to_xml) if document.children.first.name == 'XMLDataContainer'
```

Comment:
V AG sme mali s niečím podobným problémy, ak to bolo naformátované xml - boli tam whitepsaces. Potom tie first children v Jave hádzalo tie whitespaces. Skúšala si tam dať aj naformátované xdc?

Reply:
Dobra poznamka, prisposobila som aj pre tento pripad.